lib/Customer.php

```diff
@@ -78,192 +74,199 @@ public function deleteDiscount($params = null, $opts = null)
+    public static function retrieveBalanceTransaction($id, $balanceTransactionId, $params = null, $opts = null)
     {
-        return self::_retrieveNestedResource($id, static::PATH_SOURCES, $sourceId, $params, $opts);
+        return self::_retrieveNestedResource($id, static::PATH_BALANCE_TRANSACTIONS, $balanceTransactionId, $params, $opts);
     }
 
     /**
-     * @param string $id The ID of the customer to which the source belongs.
-     * @param string $sourceId The ID of the source to update.
+     * @param string $id The ID of the customer to which the customer balance transaction belongs.
+     * @param string $balanceTransactionId The ID of the customer balance transaction to update.
      * @param array|null $params
      * @param array|string|null $opts
      *
      * @throws \Stripe\Exception\ApiErrorException if the request fails
      *
-     * @return ApiResource
+     * @return BalanceTransaction
      */
-    public static function updateSource($id, $sourceId, $params = null, $opts = null)
+    public static function updateBalanceTransaction($id, $balanceTransactionId, $params = null, $opts = null)
     {
-        return self::_updateNestedResource($id, static::PATH_SOURCES, $sourceId, $params, $opts);
+        return self::_updateNestedResource($id, static::PATH_BALANCE_TRANSACTIONS, $balanceTransactionId, $params, $opts);
     }
 
     /**
-     * @param string $id The ID of the customer to which the source belongs.
-     * @param string $sourceId The ID of the source to delete.
+     * @param string $id The ID of the customer on which to retrieve the customer balance transactions.
      * @param array|null $params
      * @param array|string|null $opts
      *
      * @throws \Stripe\Exception\ApiErrorException if the request fails
      *
-     * @return ApiResource
+     * @return Collection The list of customer balance transactions.
      */
-    public static function deleteSource($id, $sourceId, $params = null, $opts = null)
+    public static function allBalanceTransactions($id, $params = null, $opts = null)
     {
-        return self::_deleteNestedResource($id, static::PATH_SOURCES, $sourceId, $params, $opts);
+        return self::_allNestedResources($id, static::PATH_BALANCE_TRANSACTIONS, $params, $opts);
     }
 
+    const PATH_SOURCES = '/sources';
+
     /**
-     * @param string $id The ID of the customer on which to retrieve the sources.
+     * @param string $id The ID of the customer on which to create the payment source.
      * @param array|null $params
      * @param array|string|null $opts
      *
      * @throws \Stripe\Exception\ApiErrorException if the request fails
      *
-     * @return Collection The list of sources.
+     * @return AlipayAccount|BankAccount|BitcoinReceiver|Card|Source
      */
-    public static function allSources($id, $params = null, $opts = null)
+    public static function createSource($id, $params = null, $opts = null)
```

Comment:
It would be great if we enforced the order of those operations. Either we alphabetize entirely so all is first, or we keep what we had). I'm not strongly opposed but I worry that we rely on the order in the spec which could change in the future, making diffs impossible to review